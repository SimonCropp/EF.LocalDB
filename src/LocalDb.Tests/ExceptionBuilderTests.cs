using System;
using ObjectApproval;
using Xunit;
using Xunit.Abstractions;

public class ExceptionBuilderTests :
    XunitLoggingBase
{
    [Fact]
    public void WrapLocalDbFailure()
    {
        var wrapped = ExceptionBuilder.WrapLocalDbFailure("InstanceName", @"c:\LocalDBData\InstanceName", new Exception());
        ObjectApprover.VerifyWithJson(wrapped);
    }
    [Fact]
    public void WrapLocalDbFailure_message()
    {
        var wrapped = ExceptionBuilder.WrapLocalDbFailure("InstanceName", @"c:\LocalDBData\InstanceName", new Exception());
        ApprovalTests.Approvals.Verify(wrapped.Message);
    }

    public ExceptionBuilderTests(ITestOutputHelper output) :
        base(output)
    {
    }
}