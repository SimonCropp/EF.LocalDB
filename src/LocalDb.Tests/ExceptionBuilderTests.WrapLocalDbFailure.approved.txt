{
  Message: 'Failed to setup a LocalDB instance.\r\nname: InstanceName\r\ndirectory: c:\LocalDBData\InstanceName:\r\n\r\nTo cleanup perform the following actions:\r\n * Execute \'sqllocaldb stop InstanceName\'\r\n * Execute \'sqllocaldb delete InstanceName\'\r\n * Delete the directory c:\LocalDBData\InstanceName\'\r\n',
  Data: {},
  InnerException: {
    Message: 'Exception of type \'System.Exception\' was thrown.',
    Data: {}
  }
}